users/views.py

```diff
@@ -0,0 +1,65 @@
+import json
+import jwt
+import re
+import bcrypt
+
+from django.shortcuts import render
+from django.views import View
+from django.core.validators import validate_email
+from django.core.exceptions import ValidationError
+from django.http import (
+    JsonResponse,
+    HttpResponse
+)
+
+from dr_martens.settings import (
+    SECRET_KEY,
+    ALGORITHM
+)
+from users.models import User
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            validate_email(data['email'])
+            if User.objects.filter(nickname = data['nickname']).exists():
+                return HttpResponse(status = 401)
+
+            if re.match("^[0-9A-Za-z]{6,20}$", data['nickname']) == None:
+                return HttpResponse(status = 401)
+
+            if re.match("^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d$@$!%*#?&]{6,20}$", data['password']) == None:
+                return HttpResponse(status = 401)
+
+            hashed_password = bcrypt.hashpw(data['password'].encode('utf-8'), bcrypt.gensalt())
+            User(
+                name         = data['name'],
+                nickname     = data['nickname'],
+                password     = hashed_password.decode('utf-8'),
+                birthday     = data['birthday'],
+                email        = data ['email'],
+                phone_number = data['phone_number'],
+            ).save()
+            return HttpResponse(status = 200)
+
+        except ValidationError:
+            return JsonResponse({'message' : 'INVALID_EMAIL'}, status = 400)
+
+        except KeyError as e:
+            return JsonResponse({'message' : str(e) + ' is right key name.'}, status = 400)
+
+class SignInView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            if User.objects.filter(nickname = data['nickname']).exists():
+                users = User.objects.get(nickname = data['nickname'])
+                if bcrypt.checkpw(data['password'].encode('utf-8'), users.password.encode('utf-8')):
+                    token = jwt.encode({'users_id' : users.id}, SECRET_KEY, algorithm = ALGORITHM)
```

Comment:
```suggestion
                    token = jwt.encode({'user_id' : user.id}, SECRET_KEY, algorithm = ALGORITHM)
```
데코레이터 코드에 expired 처리하셨는데 여기에 별도의 exp값이 없네요?

Reply:
데코레이터 코드에 expired 삭제했습니다